File: frontendapp/lib/models/bookingdetail.dart
class BookingDetail {
  final int? id;
  final int bookingId;
  final int seatId;
  final double price;

  BookingDetail({
    this.id,
    required this.bookingId,
    required this.seatId,
    required this.price,
  });

  factory BookingDetail.fromMap(Map<String, dynamic> map) => BookingDetail(
    id: map['ID'],
    bookingId: map['BOOKINGID'],
    seatId: map['SEATID'],
    price: map['PRICE'].toDouble(),
  );

  Map<String, dynamic> toMap() => {
    'ID': id,
    'BOOKINGID': bookingId,
    'SEATID': seatId,
    'PRICE': price,
  };
}

File: backend/bookingmovieticketapp/src/main/java/com/project/bookingmovieticketapp/Models/BookingDetail.java
package com.project.bookingmovieticketapp.Models;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "bookingdetails")
@Builder
public class BookingDetail {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne
    @JoinColumn(name = "bookingid")
    private Booking booking;

    @ManyToOne
    @JoinColumn(name = "seatid")
    private Seat seat;

    @Column(name = "price",nullable = false)
    private int price;
}
